Update UserGuide Memory and Undo-Redo features (#99)
AY1920S2-CS2103T-W17-3/main: docs/UserGuide.adoc

****
* This command can only be used in the `Memory` tab using the command `goto` in a `Trip`.
* `INDEX` refers to the index number shown in the displayed diary entry list. `INDEX` *must be a positive integer* 1, 2, 3, ...
* At least one of the optional fields must be provided.
* Existing values will be updated to the input values.
* When editing tags, the existing tags of the diary entry will be removed i.e adding of tags is not cumulative.
* You can remove all the diary entry's tags by typing `t/` without specifying any tags after it.
****

Example:

* `editentry 1 title/Breakfast a 5-star hotel` +
Edits the title of the 1st diary entry.
* `editentry 2 tag/`
Clears the tags of the 2nd diary entry.

==== Delete an existing text entry: `deleteentry`
Deletes an existing diary entry at the specified `INDEX`. +
Format: `deleteentry INDEX`

****
* This command can only be used in the `Memory` tab using the command `goto` in a `Trip`.
* `INDEX` refers to the index number shown in the displayed diary entry list. `INDEX` *must be a positive integer* 1, 2, 3, ...
****

Example:

* `deleteentry 10` +
Deletes the 10th diary entry.

> ==== Locate diary entry by keywords: `find`

Author: ```suggestion
==== Locate diary entries by keywords: `find`
```

> Locate existing diary entries by keywords. +

Author: ```suggestion
Locates existing diary entries by keywords. +
```

* `KEYWORD` and `MORE_KEYWORD` are case insensitive. e.g *skydive* will match *Skydive*
* The order of the keywords does not matter. e.g. *swim beach* will match *beach swim*
* Both the title and details are searched.
* Partial words will be matched e.g. *sky* will match *skydive*
* Diary entries matching at least one keyword will be returned (i.e. OR search). e.g. *skydive beach* can return *My first skydive experience* and *Swimming at Sanur Beach*
****

==== Add a new photo: addphoto
Adds a new photo to the photo album of this `Trip`.
Format: `addphoto caption/CAPTION path/RELATIVE_PATH_OF_PHOTO`

****
* This command can only be used in the `Memory` tab using the command `goto` in a `Trip`.
* `RELATIVE_PATH_OF_PHOTO` refers to the path from the directory of this application to the photo.
* `RELATIVE_PATH_OF_PHOTO` only accepts a `.png` or `.jpg` image file.
****

Example:

* `addphoto caption/Sanur Beach path/photos/SanurBeach.jpg` +
Adds an image named `SanurBeach.jpg` in the folder named `photos` in the same directory as this application
to the photo album with the caption: "Sanur Beach".

> ==== Set passcode for memories: `setpasscode`

Author: ```suggestion
==== Set passcode for the Memory tab: `setpasscode`
```

> Set a passcode for the memories tab for user to lock and unlock their data. +

Author: ```suggestion
Sets a passcode for the Memory tab for user to lock and unlock the Memory tab features. +
```

Format: `setpasscode PASSCODE`

****
* This command can only be used in the `Memory` tab using the command `goto` in a `Trip`.
* `PASSCODE` can only be made up of at most 16 alphanumeric characters.
****

[TIP]
For a secure passcode, make sure your passcode is long and contains alphanumeric characters.

[WARNING]
Passcode recovery system will only be implemented in V2.0. If the user forgets the passcode, there is no way to recovery it.

Example:
* `setpasscode l0NgP455w0Rd`

> ==== Change passcode for memories: `changepasscode`

Author: ```suggestion
==== Change passcode for the Memory tab: `changepasscode`
```

> Change the passcode for the memories tab. +

Author: ```suggestion
Changes the passcode for the Memory tab. +
```

Format: `changepasscode OLD_PASSCODE NEW_PASSCODE`

****
* This command can only be used in the `Memory` tab using the command `goto` in a `Trip`.
* This command can only be used after setting a passcode for the `Memory` tab using the command `setpasscode`.
* `OLD_PASSCODE` and `NEW_PASSCODE` can only be made up of at most 16 alphanumeric characters each.
****

Example:
* `changepasscode l0NgP455w0Rd l0Ng3rP455w0Rd`

> ==== Lock memories: `lock`

Author: ```suggestion
==== Lock the Memory tab: `lock`
```

> Locks the ability to view and edit the diary and photo album in the `Memory` tab. +

Author: ```suggestion
Locks the ability to view and edit the diary and photo albums in the `Memory` tab. +
```

Format: `lock`

****
* This command can only be used in the `Memory` tab using the command `goto` in a `Trip`.
* This command can only be used after setting a passcode for the `Memory` tab using the command `setpasscode`.
* `lock` will disable all commands related to the `Memory` tab other than the command `unlock`.
****

Example:

* `lock`

> ==== Unlock memories: `unlock`

Author: ```suggestion
==== Unlock the Memory tab: `unlock`
```

Unlocks the restrictions enforced by the command `lock`.
Format: `unlock PASSCODE`

****
* This command can only be used after opening a `Trip` using the command `open`.
* This command can only be used after the `Memory` tab is locked using the command `lock`.
* To unlock the restrictions successfully, `PASSCODE` must be equivalent to the passcode set for the `Memory` tab using the command `setpasscode`.
****

Example:

* `unlock l0Ng3rP455w0Rd` +
Used "l0Ng3rP455w0Rd" as the `PASSCODE` to unlock the restrictions enforced by the command `lock`.

==== Export the diary to pdf file [Coming in V2.0]

=== Undo-Redo

> Allows the user to revert their actions quickly and conveniently.

Author: ```suggestion
Allows the user to undo and redo their input commands quickly and conveniently.
```

> ==== Undo an action: `undo`

Author: ```suggestion
==== Undo the previous input command: `undo`
```

> Undo an undoable command entered by the user. +

Author: ```suggestion
Reverts the effect of the previous input command entered by the user. +
```

Format: `undo`

Example:

* `undo`

> ==== Redo an action: `redo`

Author: ```suggestion
==== Redo an undone input command: `redo`
```

> Redo an undone action after using the command `undo`. +

Author: ```suggestion
Re-performs an undone input command after using the command `undo`. +
```
